derive_builder_core/src/builder.rs

```diff
@@ -140,6 +140,48 @@ pub struct Builder<'a> {
     pub std: bool,
 }
 
+impl<'a> Builder<'a> {
+    #[doc = "
+        adding builder() fn on the original struct
+        ```rust
+        impl <original_struct> {
+         pub fn builder() -> <original_struct>Builder  {
+             <original_struct>Builder::default()
+         }
+        }
+        ```
+        Why ?
+        Because sometimes @alexzanderr likes to call the builder like this:
+        ```rust
+            let ch = Channel::builder()
+            .special_info(42u8)
+            .token(19_124)
+            .build()
+            .unwrap();
+        ```
+        Note the `Channel::builder()`, it generates the builder from the original struct.
+    "]
+    pub fn add_builder_fn_to_struct(&self, ast: &syn::DeriveInput) -> TokenStream {
+        // name of the original struct
+        let struct_name_ident = &ast.ident;
+        // name of the builder for the original struct
+        let builder_name_ident = &self.ident;
+
+        // rust source code
+        let expanded = quote! {
+            /// This returns a new builder for the original struct
+            impl #struct_name_ident {
+                pub fn builder() -> #builder_name_ident {
+                    #builder_name_ident::default()
```

Comment:
This should be using the standard from elsewhere in the crate of addressing the `default` method; see [this line](https://github.com/colin-kiegel/rust-derive-builder/blob/master/derive_builder_core/src/default_expression.rs#L33) for the right way to do that.

If this isn't done, a builder struct with an inherent method named `default` would run the wrong method here, as explained in The Book [here](https://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/book/first-edition/ufcs.html).